cou/steps/analyze.py

```diff
@@ -74,14 +74,20 @@ async def _populate(cls) -> list[OpenStackApplication]:
             for app, app_status in juju_status.applications.items()
         }
         # remove non-supported charms that return None on AppFactory.create
-        apps.remove(None)
-        upgradeable_apps = {app for app in apps if app and app.charm in UPGRADE_ORDER}
-        unknown_apps = apps - upgradeable_apps
-        upgradeable_apps_sorted = sorted(
-            upgradeable_apps, key=lambda app: UPGRADE_ORDER.index(app.charm)
+        apps.discard(None)
+        apps_to_upgrade_in_order = {app for app in apps if app and app.charm in UPGRADE_ORDER}
+        disordered_upgrade_apps = apps - apps_to_upgrade_in_order
+        sorted_apps_to_upgrade_in_order = sorted(
+            apps_to_upgrade_in_order, key=lambda app: UPGRADE_ORDER.index(app.charm)
+        )
+        disordered_upgrade_apps_sorted_by_name = sorted(
```

Comment:
I think you're sorting them by name so that the plan doesn't randomly change over multiple executions, right? I suggest adding a quick comment explaining why the sorting is relevant.